core/src/server/net/listener_config.hpp

```diff
@@ -29,7 +32,10 @@ struct PortConfig {
     crypto::PrivateKey tls_private_key;
     std::vector<crypto::Certificate> tls_certificate_authorities;
 
+    std::optional<crypto::SslCtx> ssl_ctx;
+
     void ReadTlsSettings(const storages::secdist::SecdistConfig& secdist);
+    void InitSslCtx();
```

Comment:
Does it make sense to have a separate method though? If they need to be called together anyway